packages/web3-core/src/web3_request_manager.ts

```diff
@@ -64,22 +70,22 @@ export class Web3RequestManager extends Web3EventEmitter<{
 		return this._providers;
 	}
 
-	public setProvider(provider: SupportedProviders | string, net?: Socket) {
-		let newProvider!: Web3BaseProvider;
+	public setProvider(provider: SupportedProviders<API> | string, net?: Socket) {
+		let newProvider!: Web3BaseProvider<API>;
 
 		// autodetect provider
 		if (provider && typeof provider === 'string' && this.providers) {
 			// HTTP
 			if (/^http(s)?:\/\//i.test(provider)) {
-				newProvider = new this.providers.HttpProvider(provider);
+				newProvider = new this.providers.HttpProvider<API>(provider);
```

Comment:
if user is passing string how <API> is used?

Reply:
The string refers to the connection string, not provider type. The `API` here refers to the generic value for the newly created type.

Reply:
thats exactly my question :) if user is passing string ( connection string ) , this code will use `API` generic type provider by default? and user will not be able to specify it via param?